### ISSUE TITLE
Generic number of traps

### ISSUE THREAD

**Issue by RemiTheWarrior** [RemiTheWarrior/FESTIM#25]
Find a way to avoid hard coding the number of simulated traps.
Replace:
```
P1 = FiniteElement('P', interval, 1)
element = MixedElement([P1, P1, P1, P1])
V = FunctionSpace(mesh, element)
W = FunctionSpace(mesh, 'P', 1)
V0 = FunctionSpace(mesh, 'DG', 0)
```
```
v_1, v_2, v_3, v_4 = TestFunctions(V)
testfunctions = [v_1, v_2, v_3, v_4]
```
```
u_1, u_2, u_3, u_4 = split(u)
solutions = [u_1, u_2, u_3, u_4]
```
```
ini_u = Expression(("0", "0", "0", "0"), degree=1)
u_n = interpolate(ini_u, V)
u_n1, u_n2, u_n3, u_n4 = split(u_n)
previous_solutions = [u_n1, u_n2, u_n3, u_n4]

```

```
    _u_1, _u_2, _u_3, _u_4 = u.split()

    # Save solution to file (.xdmf)
    _u_1.rename("solute", "label")
    _u_2.rename("trap_1", "label")
    _u_3.rename("trap_2", "label")
    _u_4.rename("trap_3", "label")
    xdmf_u_1.write(_u_1, t)
    xdmf_u_2.write(_u_2, t)
    xdmf_u_3.write(_u_3, t)
    xdmf_u_4.write(_u_4, t)

    total_trap1 = assemble(_u_2*dx)
    total_trap2 = assemble(_u_3*dx)
    total_trap3 = assemble(_u_4*dx)
    total_trap = total_trap1 + total_trap2 + total_trap3
    total_sol = assemble(_u_1*dx)
```

**Comment by RemiTheWarrior** [comment]
⚠️ When less than 5 traps are set, calculation yields `nan`. 

Workarround is to set traps with 0 density.

**Comment by RemiTheWarrior** [comment]
Bug fixed in #64.
